tailwarden/komiser
# feat: display dependency graph on the "Resource Explorer" page

## Problem

Added the dependency graph feature

## Solution

We're now rendering the dependency graph with [Cytoscapejs](https://js.cytoscape.org/)

## Changes Made

- add [cytoscapejs](https://js.cytoscape.org/)
- add API calls to fetch resources and their mappings/relations
- update filter to be reused (somewhat) for the dependency graph

## How to Test

Check out the repo, run `npm install`, start the backend and frontend server and visit http://localhost:3002/explorer/

## Screenshots

![Screenshot 2023-09-27 at 15 15 50](https://github.com/tailwarden/komiser/assets/320272/0c64fc4d-f002-4686-a560-3c5c7ec98fc3)
![Screenshot 2023-09-27 at 15 15 58](https://github.com/tailwarden/komiser/assets/320272/bb9c98c7-26c8-42ba-af9c-a146f9abae79)
![Screenshot 2023-09-27 at 15 16 34](https://github.com/tailwarden/komiser/assets/320272/40fb88d0-663f-4902-a00c-6b47baa557d1)
![Screenshot 2023-09-27 at 15 16 41](https://github.com/tailwarden/komiser/assets/320272/eae56056-31fe-4239-b7a8-f5eb4abb7230)
![Screenshot 2023-09-27 at 15 17 09](https://github.com/tailwarden/komiser/assets/320272/1c676bec-4cb5-4bdc-8952-c0b14d9832b0)


## Notes

Since Inventory and Explorer now both share the Filter component we should refactor it someday in the future. I did not include it in this PR and only did the required changes for getting it to work!

## Checklist


## Reviewers

see reviewer list in the sidebar
